bigtable/google/cloud/bigtable/instance.py

```diff
@@ -241,23 +241,41 @@ def create(self, location_id=_EXISTING_INSTANCE_LOCATION_ID,
             clusters=clusters)
 
     def update(self):
-        """Update this instance.
+        """Updates an instance within a project.
 
         .. note::
 
-            Updates the ``display_name``. To change that value before
-            updating, reset its values via
+            Apdates any or all of the following values:
+            ``display_name``
+            ``type``
+            ``labels``
+            To change a value before
+            updating, reset that values via
 
             .. code:: python
 
                 instance.display_name = 'New display name'
 
             before calling :meth:`update`.
+
+        :rtype: :class:`~google.api_core.operation.Operation`
+        :returns: The long-running operation corresponding to the update
+                    operation.
         """
-        self._client.instance_admin_client.update_instance(
+        update_mask_pb = field_mask_pb2.FieldMask()
+        if self.display_name is not None:
+            update_mask_pb.paths.append('display_name')
+        if self.type_ is not None:
+            update_mask_pb.paths.append('type')
+        if self.labels is not None:
+            update_mask_pb.paths.append('labels')
+        instance_pb = instance_pb2.Instance(
             name=self.name, display_name=self.display_name,
-            type_=self.type_,
-            labels=self.labels)
+            type=self.type_, labels=self.labels)
+
+        return self._client.instance_admin_client.partial_update_instance(
+            instance=instance_pb,
+            update_mask=update_mask_pb)
```

Comment:
Does a [`PartialUpdateInstance` call](https://cloud.google.com/bigtable/docs/reference/admin/rpc/google.bigtable.admin.v2#google.bigtable.admin.v2.PartialUpdateInstanceRequest) not return anything with the operation metadata that we need to process?

Reply:
I don't think that there's anything that needs to be processed.  The LongRunningOperation is just a way to make the RPC return a bit faster.  `update_instance` returned the `Instance` right away, but after the Cloud Bigtable replication feature the method returned slower, hence the LongRunningOperation.

Reply:
OK.